Discuss some more aspects of 2d models. (#6459)
geodynamics/aspect: doc/sphinx/user/methods/basic-equations/2d-models.md

If one adopts this point of view, the Stokes equations {math:numref}`eq:stokes-1`-{math:numref}`eq:stokes-2` naturally simplify in a way that allows us to reduce the $3+1$ equations to only $2+1$, but it makes clear that the correct description of the compressible strain rate is still $\varepsilon(\mathbf u) - \frac{1}{3}(\nabla \cdot \mathbf u)\mathbf 1$, rather than using a factor of $\frac{1}{2}$ for the second term.
(A derivation of why the compressible strain rate tensor has this form can be found in {cite:t}`schubert:etal:2001`, sec. 6.5.)

Because we think of these 2d models as just a slice through an infinite cylinder along the $z$-axis, we should still see what happens if we express the strain rate as a $3\times 3$ tensor. Under the assumptions above, we have
```{math}
\begin{aligned}
  \varepsilon(\mathbf u) = \begin{pmatrix} \tfrac{\partial u_x}{\partial x} & \tfrac 12 \tfrac{\partial u_x}{\partial y} + \tfrac 12 \tfrac{\partial u_y}{\partial x} & 0 \\
    \tfrac 12 \tfrac{\partial u_x}{\partial y} + \tfrac 12 \tfrac{\partial u_y}{\partial x} & \tfrac{\partial u_y}{\partial y} & 0 \\
    0 & 0 & 0 \end{pmatrix}
\end{aligned}
```

> with the expected zeros in the last row and column: the last row contains derivatives of $u_z=0$, whereas the last column contains $z$-derivatives of all velocities (which are also zero under our assumption). Because the strain rate tensor is zero all $z$-related entries, our assumption about what 2d models represent is often calls "[plane strain](https://en.wikipedia.org/wiki/Infinitesimal_strain_theory#Plane_strain)" in mechanics, indicating that nonzero strain components are all in the $x$-$y$ plane. Interestingly, though, for compressible materials, the full compressible strain rate tensor then reads

Reviewer: ```suggestion
with the expected zeros in the last row and column: the last row contains derivatives of $u_z=0$, whereas the last column contains $z$-derivatives of all velocities (which are also zero under our assumption). Because the strain rate tensor is zero for all $z$-related entries, our assumption about what 2d models represent is often called "[plane strain](https://en.wikipedia.org/wiki/Infinitesimal_strain_theory#Plane_strain)" in mechanics, indicating that nonzero strain components are all in the $x$-$y$ plane. Interestingly, though, for compressible materials, the full compressible strain rate tensor then reads
```

The entry in the $(3,3)$ position of this tensor may be surprising.
It disappears, however, when taking the (three-dimensional) divergence of the stress, as is done in {math:numref}`eq:stokes-1`, because the divergence applies the $z$ derivative to all elements of the last row - and the assumption above was that all $z$ derivatives are zero; consequently, whatever lives in the third row of the strain rate tensor does not matter.

Two other comments are in order:

> * Above, we have described how we think about 2d models, namely in the "plane strain" mental model. One could, however, come up with other thoughts. Specifically, we could have assumed, for example, that where heating occurs, the material expands not only in the $x$- and $y$-directions, but isotropically also into the third dimension. In such cases, one would also obtain velocities that have a $z$-component, typically in such a way that $u_z$ is *proportional* to the distance from the $x$-$y$-plan in which we simulate the material. This then results in models that we would call "[plane stress](https://en.wikipedia.org/wiki/Plane_stress)". We specifically do not adopt this perspective.

Reviewer: ```suggestion
* Above, we have described how we think about 2d models, namely in the "plane strain" mental model. One could, however, come up with other thoughts. Specifically, we could have assumed, for example, that where heating occurs, the material expands not only in the $x$- and $y$-directions, but isotropically also into the third dimension. In such cases, one would also obtain velocities that have a $z$-component, typically in such a way that $u_z$ is *proportional* to the distance from the $x$-$y$-plane in which we simulate the material. This then results in models that we would call "[plane stress](https://en.wikipedia.org/wiki/Plane_stress)". We specifically do not adopt this perspective.
```

> * Above, we have described how we think about 2d models, namely in the "plane strain" mental model. One could, however, come up with other thoughts. Specifically, we could have assumed, for example, that where heating occurs, the material expands not only in the $x$- and $y$-directions, but isotropically also into the third dimension. In such cases, one would also obtain velocities that have a $z$-component, typically in such a way that $u_z$ is *proportional* to the distance from the $x$-$y$-plane in which we simulate the material. This then results in models that we would call "[plane stress](https://en.wikipedia.org/wiki/Plane_stress)". We specifically do not adopt this perspective.

Reviewer 2: ```suggestion
* Above, we have described how 2d models are implemented in ASPECT, namely using the "plane strain" approximation. There are, however, other approximations. Specifically, we could have assumed, for example, that where changes in temperature or pressure occur, the resulting change in volume is partitioned in a 2:1 ratio between the $x-y$-plane and $z$-dimension. In this approximation, one would obtain nonzero velocities $u_z$ that are *proportional* to the distance from the $x$-$y$-plane in which we simulate the material. This is related to the "[plane stress](https://en.wikipedia.org/wiki/Plane_stress)" approximation (which, strictly speaking, has $\sigma_{iz}=0$). We specifically do not adopt this perspective.
```

> * It is interesting to consider what happens when we think of the plane strain model if the material is incompressible. Incompressibility of the conceptual 3d object implies that $\nabla \cdot \vec u = \frac{\partial u_x}{\partial x} + \frac{\partial u_y}{\partial y} + \frac{\partial u_z}{\partial z}=0$. Under our assumptions, the last term is zero, and as a consequence, in our two-dimensional cross-section of the infinite three-dimensional domain, we also have $\nabla_{2d} \cdot \vec u_2d} = \frac{\partial u_x}{\partial x} + \frac{\partial u_y}{\partial y}=0$; i.e., the material is also incompressible in the 2d cross section. Perhaps equally importantly, if a material expands its three-dimensional volume by a factor of $\alpha$, under the plane strain assumption, it must also expand its two-dimensional (cross-sectional) area by a factor of $\alpha$. A second consequence is that if you consider the full strain tensor for incompressible materials, i.e., $\varepsilon(\mathbf u) - \frac{1}{3}(\nabla \cdot \mathbf u)\mathbf 1$, then under the plane strain assumption, it does not actually matter whether one uses $\frac{1}{3}$ of $\frac{1}[2}$ (as one might naively do for two-dimensional models): $\nabla \cdot \mathbf u$ is zero, and so the factor we multiply that term by is unimportant. Of course, if the material *is* compressible, then $\nabla \cdot \mathbf u$ is in general not zero, and there *is* a difference between the two choices for the factor. In those cases, it is important to be clear what notion we adopt for "2d problem"; the assumptions we have laid out above then imply that even for 2d problems, the choice of $\frac{1}{3}$ is correct.

Reviewer: Maybe expand this a bit to be very clear about the case that is handled incorrectly in some material models:

```suggestion
* It is interesting to consider what happens when we think of the plane strain model if the material is incompressible. Incompressibility of the conceptual 3d object implies that $\nabla \cdot \vec u = \frac{\partial u_x}{\partial x} + \frac{\partial u_y}{\partial y} + \frac{\partial u_z}{\partial z}=0$. Under our assumptions, the last term is zero, and as a consequence, in our two-dimensional cross-section of the infinite three-dimensional domain, we also have $\nabla_{2d} \cdot \vec u_2d} = \frac{\partial u_x}{\partial x} + \frac{\partial u_y}{\partial y}=0$; i.e., the material is also incompressible in the 2d cross section. Perhaps equally importantly, if a material expands its three-dimensional volume by a factor of $\alpha$, under the plane strain assumption, it must also expand its two-dimensional (cross-sectional) area by a factor of $\alpha$. A second consequence is that if you consider the full strain tensor for incompressible materials, i.e., $\varepsilon(\mathbf u) - \frac{1}{3}(\nabla \cdot \mathbf u)\mathbf 1$, then under the plane strain assumption, it does not actually matter whether one uses $\frac{1}{3}$ or $\frac{1}[2}$ (as one might naively do for two-dimensional models): $\nabla \cdot \mathbf u$ is zero, and so the factor we multiply that term by is unimportant. Of course, if the material *is* compressible, then $\nabla \cdot \mathbf u$ is in general not zero, and there *is* a difference between the two choices for the factor. In those cases, it is important to be clear what notion we adopt for "2d problem"; the assumptions we have laid out above then imply that even for 2d problems, the choice of $\frac{1}{3}$ is correct and should for example be used when computing a deviatoric strain rate for any purpose.
```

> * It is interesting to consider what happens when we think of the plane strain model if the material is incompressible. Incompressibility of the conceptual 3d object implies that $\nabla \cdot \vec u = \frac{\partial u_x}{\partial x} + \frac{\partial u_y}{\partial y} + \frac{\partial u_z}{\partial z}=0$. Under our assumptions, the last term is zero, and as a consequence, in our two-dimensional cross-section of the infinite three-dimensional domain, we also have $\nabla_{2d} \cdot \vec u_2d} = \frac{\partial u_x}{\partial x} + \frac{\partial u_y}{\partial y}=0$; i.e., the material is also incompressible in the 2d cross section. Perhaps equally importantly, if a material expands its three-dimensional volume by a factor of $\alpha$, under the plane strain assumption, it must also expand its two-dimensional (cross-sectional) area by a factor of $\alpha$. A second consequence is that if you consider the full strain tensor for incompressible materials, i.e., $\varepsilon(\mathbf u) - \frac{1}{3}(\nabla \cdot \mathbf u)\mathbf 1$, then under the plane strain assumption, it does not actually matter whether one uses $\frac{1}{3}$ or $\frac{1}[2}$ (as one might naively do for two-dimensional models): $\nabla \cdot \mathbf u$ is zero, and so the factor we multiply that term by is unimportant. Of course, if the material *is* compressible, then $\nabla \cdot \mathbf u$ is in general not zero, and there *is* a difference between the two choices for the factor. In those cases, it is important to be clear what notion we adopt for "2d problem"; the assumptions we have laid out above then imply that even for 2d problems, the choice of $\frac{1}{3}$ is correct and should for example be used when computing a deviatoric strain rate for any purpose.

Reviewer 2: ```suggestion
* It is interesting to consider what happens when we think of the plane strain model if the material is incompressible. Incompressibility of the conceptual 3d object implies that $\nabla \cdot \vec u = \frac{\partial u_x}{\partial x} + \frac{\partial u_y}{\partial y} + \frac{\partial u_z}{\partial z}=0$. Under our assumptions, the last term is zero, and as a consequence, in our two-dimensional cross-section of the infinite three-dimensional domain, we also have $\nabla_{2d} \cdot \vec u_2d} = \frac{\partial u_x}{\partial x} + \frac{\partial u_y}{\partial y}=0$; i.e., the material is also incompressible in the 2d cross section. Perhaps equally importantly, if a material expands its three-dimensional volume by a factor of $\alpha$, under the plane strain assumption, it must also expand its two-dimensional (cross-sectional) area by a factor of $\alpha$. A second consequence is that if you consider the full strain tensor for incompressible materials, i.e., $\varepsilon(\mathbf u) - \frac{1}{3}(\nabla \cdot \mathbf u)\mathbf 1$, then under the plane strain assumption, it does not actually matter whether one uses $\frac{1}{3}$ or $\frac{1}[2}$ (as one might naively do for two-dimensional models): $\nabla \cdot \mathbf u$ is zero, and so the factor we multiply that term by is unimportant. Of course, if the material *is* compressible, then $\nabla \cdot \mathbf u$ is in general not zero, and there *is* a difference between the two choices for the factor. In those cases, it is important to be clear what notion we adopt for the "2d problem"; the assumptions we have laid out above then imply that even for 2d problems, the choice of $\frac{1}{3}$ is correct and should for example be used when computing a deviatoric strain rate for any purpose.
```

# Comments on the final set of equations

> ASPECT solves the equations stated in the previous sections in essentially the form stated.

Reviewer 2: ```suggestion
ASPECT solves the equations given in the previous sections in essentially the form stated.
```
